VETTING: Disable "vetting by phone" if the requirements are not meet

eunjuhuss opened the pull request:
#### Description:
Disable "vetting by phone" if the phone number is either:
- not added,
- if the number is not confirmed
- or if the number is not a Swedish phone number. 

#### Summary:
- only remaining modal for verify identity 
- disable button if no added phone number, not confirmed phone number or not a Swedish number
- edited the help text under the button, to be gray colour when the button is disabled
- removed unused translated texts
- add translated texts
  - Start by adding your phone number in settings / Börja med att lägga till ditt telefonnummer under inställningar
  - Confirm your phone number in settings / Bekräfta ditt telefonnummer under inställningar
  - Only possible with Swedish phone number / Bara möjligt med svenskt telefonnummer
---
- If no phone number is added
<img width="1439" alt="Screenshot 2020-12-16 at 15 09 48" src="https://user-images.githubusercontent.com/44289056/102359200-c5c8d780-3fb0-11eb-914e-6e8bc586ecba.png">



- If the phone number is not confirmed
<img width="1433" alt="Screenshot 2020-12-16 at 15 10 58" src="https://user-images.githubusercontent.com/44289056/102359317-ee50d180-3fb0-11eb-888f-fe72630fb301.png">


- If confirmed Swedish phone number
<img width="1430" alt="Screenshot 2020-12-16 at 14 37 07" src="https://user-images.githubusercontent.com/44289056/102358841-489d6280-3fb0-11eb-9169-faf216abb10f.png">

- If confirmed but not a Swedish phone number 
<img width="1430" alt="Screenshot 2020-12-16 at 15 16 03" src="https://user-images.githubusercontent.com/44289056/102359924-aed6b500-3fb1-11eb-9b5a-5f8a5ad6a447.png">

---
[For future reference] I will add a link to the button to redirect the user to "settings" for adding/confirming telephone numbers

For testing with non swedish number, change +46->eg.+36 " `nonSweNumber: !state.phones.phones.some((num) => num.number.includes(+46)),`" in containers/LookupMobileProofing.js
#### For reviewer:
- [ ] Read the above description
- [ ] Reviewed the code changes
- [ ] Navigate to the branch (pulled the latest version)
- [ ] Run the code and been able to execute the expected function
- [ ] Check any styling on both desktop and mobile sizes

njons commented:
The process of adding a phone number for vetting is complicated. I think, in its current state, the instructions to the user of how to navigate this process are too subtle in this solution, so I would suggest we increase the clarity for our users.

Complication 1: Phone numbers need to be added AND verified to be usable for vetting  
- In the current solution, there is very little visual change between not having a number and having an unverified number (the  button looks the same and the amount of text is similar (although, I know the text is different))
- Can we, by visual means, make the barrier between adding and verifying a number smaller?
- Are we able to address the general confusion users express (eg support email from today) regarding when a number is "added", a number is "registered", a number is "found" by separating the process of adding/verifying a number and using it for vetting (or by making them closer)?
- Are there tricks we can use in settings to help our users better understand the process? 

Complication 2: The process requires navigating between two (to the user) unrelated places
- in the current solution, the user is told to add a number and when they do and return they are told that they have to go back
- Is there a visually more impactful way top let the user know their phone number is not ready for vetting?
- Is it possible to make it clear for the user before they move to settings that they need to complete 2 steps there?
- Is an overhaul of the number vetting process justified?

eunjuhuss commented:
> The process of adding a phone number for vetting is complicated. I think, in its current state, the instructions to the user of how to navigate this process are too subtle in this solution, so I would suggest we increase the clarity for our users.

Ok, I can agree that it might be a little subtile, however, we have not change the “add phone number” process in this PR.
I have made a new design that should be more explanatory… please look through the Figma and comment.
https://www.figma.com/file/IBBTDRxd5dKQa7Ltj0Hd0l/eduID%3A-website-user-flow?node-id=814%3A5

> Complication 1: Phone numbers need to be added AND verified to be usable for vetting
> 
> In the current solution, there is very little visual change between not having a number and having an unverified number (the button looks the same and the amount of text is similar (although, I know the text is different))

This particular issue has nothing to do with this PR, I think everyone was in agreement when we added this “gray out” function, see PR453 “https://github.com/SUNET/eduid-front/pull/453” disabled without nin, I can agree that this might need adjusting however that adjustment should be separate not a part of this PR

> Can we, by visual means, make the barrier between adding and verifying a number smaller?
I have updated the design in Figma. I have add a step by step design that shows all steps from the beginning. First the user need to add nin. Second the user need to add a telephone number and last the user need to verify the telephone number. In the event of a user adding a “non Swedish phone number” there is a separat design. Please check my new Figma

> Are we able to address the general confusion users express (eg support email from today) regarding when a number is "added", a number is "registered", a number is "found" by separating the process of adding/verifying a number and using it for vetting (or by making them closer)?

Please check my new Figma

> Are there tricks we can use in settings to help our users better understand the process?

Please check my new Figma, I still think that this is outside the scope of this PR.

> Complication 2: The process requires navigating between two (to the user) unrelated places
> 
> in the current solution, the user is told to add a number and when they do and return they are told that they have to go back

We might have to adjust the setting page to clarify the verification of phone number. Maybe by adding additional information in the “notification” or the information text in “Mobile phone numbers”. 

> Is there a visually more impactful way top let the user know their phone number is not ready for vetting?

Please check my new Figma, I think that this issue is addressed in the new design.


> Is it possible to make it clear for the user before they move to settings that they need to complete 2 steps there?

Please check my new Figma, I think that this issue is addressed in the new design.

> Is an overhaul of the number vetting process justified?

Please explain what you mean by this?
The “vetting by phone” button upgrade is to align with “vetting by post” upgrade, the entire vetting process was not a part of the PR scope.

I think that we will need an other way to addres issues that is not a part of a ongoing PR’s work-scope, but that we want to adress and add to a “to do list”
What do you think?

njons reviewed (approved):
👍
